src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -127,6 +127,7 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
     void assertOnDisk() const;
 
     void remove(bool keep_s3 = false) const;
```

Comment:
There should not be any words about S3 in MergeTree.
We have similar logic for S3, HDFS and other decoupled storages.

Reply:
I agree, but we already have S3-specific flags and methods in MergeTree and it's not obvious how to remove it.

Reply:
We could rename it to more generic name like `keep_shared_data_in_decoupled_storage`, but "shared data" does not make sense for non-replicated MergeTree anyway.